File: acmicpc/15657.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;
#define MAX_LEN 9

int n, m;
int num[MAX_LEN];
vector<int> ans;

void dfs()
{
	if (ans.size() == m) {
		for (int i=0; i<ans.size(); ++i)
			cout << ans[i] << " ";
		cout << '\n';
		return;
	}

	for (int i=0; i<n; ++i) {
		if (ans.size() == 0 || ans.back() <= num[i]) {
			ans.push_back(num[i]);
			dfs();
			ans.pop_back();
		}
	}
}

int main()
{
	cin >> n >> m;
	for (int i=0; i<n; ++i)
		cin >> num[i];

	sort(num, num+n);

	dfs();

	return 0;
}

File: acmicpc/15657.py
n, m = input().split(' ')
l = input().split(' ')
l = [int(i) for i in l]
l.sort()
v = []

def dfs():
    if len(v) == int(m):
        for i in v:
            print(i, end=' ')
        print()
        return
    for i in l:
        if len(v) == 0:
            v.append(i)
            dfs()
            v.pop()
        elif v[-1] <= i:
            v.append(i)
            dfs()
            v.pop()

dfs()
